Lanchonete/Lanchonete/ Features/Customization/Presentation/CustomizationViewController.swift

```diff
@@ -9,14 +9,20 @@ import UIKit
 
 class CustomizationViewController: UITableViewController, CustomizationView {
     
-    var presenter: CustomizationViewPresenter!
+    @IBOutlet weak var confirmButton: UIButton!
+    var presenter: CustomizationViewPresenter?
     
     func reloadData() {
         tableView.reloadData()
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        let flag = presenter!.isConfirmButtonEnabled()
+        confirmButton.isEnabled = flag
+        if flag == false{
+            confirmButton.alpha = 0.5
+        }
     }
```

Comment:
Here I would suggest having plain methods invoked by the presenter instead of asking for information and taking action. For example, here you could have simple  and straightforward methods to display the button:
```
func display(confirm isEnabled: Bool) {
    confirmButton.isEnabled = isEnabled
    confirmButton.alpha = isEnabled ? 1 : 0.5
}
```
And the presenter should have a `viewDidLoad()` method (as you did in other presenters) to call the method above accordingly. Ex:
```
func viewDidLoad() {
    // Move isConfirmButtonEnabled() here and decides:
    view?.display(confirm: true)
    // or
    view?.display(confirm: false)
}
```
Note: If you noticed I used "1" in the method above because there's no case in your implementation that changes alpha again in the case of true flag.

Reply:
Thank you Marcio, I added your suggestions